src/providers/http_provider.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from abc import ABC
+from http import HTTPStatus
+from typing import Optional, Tuple
+from urllib.parse import urljoin, urlparse
+
+from requests import Session, JSONDecodeError
+from requests.adapters import HTTPAdapter
+from urllib3 import Retry
+
+
+logger = logging.getLogger(__name__)
+
+
+class NotOkResponse(Exception):
+    status: int
+    text: str
+
+    def __init__(self, *args, status: int, text: str):
+        self.status = status
+        self.text = text
+        super().__init__(*args)
+
+
+class HTTPProvider(ABC):
+    REQUEST_TIMEOUT = 300
```

Comment:
i would suggest to move it to env variables

Reply:
https://www.notion.so/Move-REQUEST_TIMEOUT-to-vars-b810954c2de74de583dda1baeac495f0?pvs=4